### PR TITLE
Add ESLint, code stye polish

### PR DESCRIPTION
**Pull request by SergioCrisostomo**

### CODE REVIEW

File: Tests/eslint-configure/eslint.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+        "curly": 0,
+        "eqeqeq": 0,
+        "eqnull": 0,
+        "nonew": 0,
+		"camelcase": 0,
+		"semi": 2,
+		"no-trailing-spaces": 2,
+		"whitespace-mootools-codestyle" : 2
+    }
+}
```
Comment (arian) [inline Tests/eslint-configure/eslint.json]: Check the codestyle of this json file ;)

File: Tests/eslint-configure/rules/no-alert.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+'use strict';
+module.exports = function (context){
+	return {
+		CallExpression: function (node){
+			if (node.callee.name === 'alert'){
+				context.report(node, 'No alert in source code please.');
+			}
+		}
+	};
+};
```
Comment (arian) [inline Tests/eslint-configure/rules/no-alert.js]: Why have this rule here?

File: Gruntfile.js [hunk 3]
```diff
@@ -8,6 +8,14 @@ module.exports = function(grunt) {
 	var pullRequest = process.env.TRAVIS_PULL_REQUEST;
 
 	grunt.initConfig({
+	    eslint: {
```
Comment (arian) [inline Gruntfile.js]: tabs/spaces.

File: Gruntfile.js [hunk 4]
```diff
@@ -8,6 +8,14 @@ module.exports = function(grunt) {
...
+			target: ['Source/**/*.js'],
```
Comment (arian) [inline Gruntfile.js]: I'd add `Gruntfile.js, Tests/**/*.js`

**Comment by anutron** [comment]
I'm going to get crucified for this, but it's 2015. Can we please switch to spaces for indentation now?

**Comment by timwienk** [comment]
No.

**Comment by arian** [comment]
and the `) {` space too?

**Comment by anutron** [comment]
Github is the normalizer here. It treats tabs as 4 spaces. Editors are smart enough that we - programmers - don't even notice which one it is. But gitub does. It makes all our lines longer than the width of the column here.

File: Gruntfile.js [hunk 5]
```diff
@@ -8,6 +8,14 @@ module.exports = function(grunt) {
...
+			options: {
+				configFile: 'Tests/eslint-configure/eslint.json',
```
Comment (arian) [inline Gruntfile.js]: If you use .eslintrc, the file is automatically picked up by the command line tool, which is useful for editor plugins for example.

**Comment by timwienk** [comment]
> and the `) {` space too?

That thing I never really cared about, that's been there because Valerio once upon a time decided he liked it. It's just been moo style since the start.

> Github is the normalizer here.

Then perhaps Github should be configurable for this as well. I'm sure they accept Pull Requests. :wink: 

> Editors are smart enough that we - programmers - don't even notice which one it is.

That makes no sense, in an editor you can obviously change tabspace, but no one in their right mind is going to change space-space.

**Comment by timwienk** [comment]
Either way, let's leave this discussion out of this PR.

**Comment by SergioCrisostomo** [comment]
Updated! I will send a similar PR to Core also.

**Comment by SergioCrisostomo** [comment]
Can we merge this?

**Comment by kentaromiura** [comment]
@anutron if it's what bother you, just add ?ts=2 at the end of the file you're reviewing, there are also browser extension that do this to all the files :)

**Pull request merged**